modules/internal/ChapelArray.chpl

```diff
@@ -3957,6 +3960,17 @@ module ChapelArray {
       a <=> b;
   }
 
+  // Sometimes we really need the result of '!' to be an array.
+  inline proc postfix!(A: [?D]) {
+    if isNilableClassType(A.eltType) {
+      var B = forall a in A do a!;
+      return B;
+    } else {
+      // no changes are necessary
```

Comment:
I would expect this to be an error for e.g. array of int. For non-nilable class type it should return an array of borrows.